test-tap/assert.js

```diff
@@ -120,8 +128,7 @@ function eventuallyFailsWith(t, fn, subset) {
 
 function fails(t, fn) {
 	lastFailure = null;
-	fn();
-	if (lastFailure) {
+	if (fn() === false && lastFailure) {
 		t.pass();
```

Comment:
Should this be something like:

```js
const retval = fn();
if (lastFailure) {
  t.false(retval)
}
```

And something similar in `passes` below. Combining the variable check with the return value check makes it hard to know what's being tested.